jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/Response.java

```diff
@@ -188,7 +188,7 @@ interface AsyncContentListener extends ContentSourceListener
         @Override
         default void onContentSource(Response response, Content.Source contentSource)
         {
-            Runnable demandCallback = Invocable.from(Invocable.InvocationType.NON_BLOCKING, () -> onContentSource(response, contentSource));
+            Runnable demandCallback = Invocable.from(Invocable.getInvocationType(contentSource), () -> onContentSource(response, contentSource));
```

Comment:
There should probably be a comment explaining why we use the invocation type of the content source: how this trick exactly works is easy to forget.

Reply:
There is no other object we can ask the invocation type to.